OD-594 Handle rating as rid edge case

hlieberman opened the pull request:
Rarely, eFiction will use the ratings themselves as the rid rather than
the number.  If we don't handle this case in conversion, no stories will
successfully be tagged with the rating.

ps-id: 71E667E9-7F7E-4D53-A385-576B3C606D96

hlieberman committed:
Handle rating as rid edge case

Rarely, eFiction will use the ratings themselves as the rid rather than
the number.  If we don't handle this case in conversion, no stories will
successfully be tagged with the rating.

ps-id: 71E667E9-7F7E-4D53-A385-576B3C606D96

hlieberman committed:
Handle rating as rid edge case

Rarely, eFiction will use the ratings themselves as the rid rather than
the number.  If we don't handle this case in conversion, no stories will
successfully be tagged with the rating.

ps-id: 71E667E9-7F7E-4D53-A385-576B3C606D96

ariana-paris commented on efiction/metadata.py:
```diff
@@ -27,6 +27,7 @@ def __init__(self, config: ConfigParser, logger: Logger, sql: SqlDb, step_path:
         self.tag_converter = TagConverter(config, logger, sql)
         self.authors = []
         self.ratings = []
+        self.ratings_dirty: bool = False
```
Haha, given the nature of some of the contents we handle, and that we wanted to move away from terminology like "sausage machine", maybe there's a better name to use here. Perhaps simply `rating_is_string`? (and change the naming in the TagConverter too)

ariana-paris replied:
And forgot to say that this could also do with a unit test to make sure it works, and doesn't regress when we next tinker with the code.

ariana-paris reviewed (commented):
Definitely sounds better! Since this repo does have automated tests, it would be good to have a couple of tests, one to check that the method correctly identifies numeric values and another to check it handles string fields. (There's probably an existing test that could be copied and adapted)

ariana-paris committed:
OD-594 Add tests for string ratings

hlieberman committed:
Merge branch 'string_ratings' of https://github.com/otwcode/open-doors-eFiction

hlieberman committed:
Explicitly cast function to return a bool

If count is an empty array, it will return that empty array instead of a
boolean. Though [] is falsy, it's a violation of the function signature.
